File: src/lessons/welcome/slug/hunting/SlugHuntingEntity.java
package lessons.welcome.slug.hunting;

import java.awt.Color;


public class SlugHuntingEntity extends jlm.universe.bugglequest.SimpleBuggle {

	@Override
	public void run() {
		hunt(); 
	}

	/* BEGIN TEMPLATE */
	public void hunt() {
		// Write your code here
		/* BEGIN SOLUTION */
		while (! isOverBaggle()) {
			if (isFacingTrail()) {
				brushDown();
				forward();
				brushUp();
			} else {
				turnLeft();
			}
		}
		pickUpBaggle();
		/* END SOLUTION */
	}

	/* BEGIN HIDDEN */
	boolean isFacingTrail() {
		if (isFacingWall())
			return false;

		forward();
		boolean res = getGroundColor().equals(Color.green);
		backward();
		return res;

	}		
	/* END HIDDEN */
	/* END TEMPLATE */


}

File: src/lessons/welcome/slug/hunting/SlugHuntingEntity.py
import java.awt.Color as Color

def isFacingTrail(trailColor):
   if isFacingWall():
      return False
   else:
	  forward()
	  res = getGroundColor().toString() == trailColor.toString()
	  backward()
	  return res

# BEGIN TEMPLATE

def hunt():
   # write your code here
   # BEGIN SOLUTION
   while not isOverBaggle():
      brushUp()
      if isFacingTrail(Color.green):
         brushDown()
         forward()
         brushUp()
      else:
	     turnLeft()
   pickUpBaggle()
   # END SOLUTION
# END TEMPLATE

hunt()
